### ISSUE TITLE
Angular application not detected. (Angular Devtools)

### ISSUE THREAD

**Issue by tpiros** [angular/angular#53953]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

I created a sample project via `ng new` and opening Angular Devtools in Chrome states that "Angular application not detected." I have not added anything nor changed the default boilerplate that gets produces after `ng new` so we are talking about a pristine app.

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.10
Node: 21.5.0 (Unsupported)
Package Manager: npm 9.8.1
OS: darwin arm64

Angular: 17.0.9
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.10
@angular-devkit/build-angular   17.0.10
@angular-devkit/core            17.0.10
@angular-devkit/schematics      17.0.10
@angular/cli                    17.0.10
@schematics/angular             17.0.10
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.3
```

**Comment by JeanMeche** [comment]
I cannot reproduce it with the same latest dependencies.  
Do you mind sharing on github the project create by the CLI ?

**Comment by sohrabi** [comment]
@tpiros 
Please provide more information.
Let us know if you see this message after you created a new Angular project using `ng new`

```
D:\test\sample>ng new sample
.
.
.
.

√ Packages installed successfully.
    Successfully initialized git.
D:\projects>
`````
If no it means one of your package is missing during installation, try run `npm i` in your project folder.
Aslo as @JeanMeche mentioned it will be really helpful if you could provide us your code in `GitHub` or `stackblitz`.

**Comment by jamesdanielspassle** [comment]
I experience the same problem. Generally it works, but often it fails. I can resolve by right-clicking inside the Angular dev tools panel and selecting "reload frame". Although this is obviously a pain.

I'm using the same app when it fails as when it works. Very strange!

**Comment by aaubry** [comment]
I'm facing the same issue with Firefox. I've just installed the devtools on Chrome and for the same site they work, but on Firefox I get "Angular application not detected. ".

**Comment by andreescocard** [comment]
same here

**Comment by dgp1130** [comment]
I took a quick look at this and I'm able to reproduce, though inconsistently. It seems like a race condition where the injected content script and the DevTools panel aren't connecting correctly, likely from recent addition of [iframe support](https://github.com/angular/angular/pull/53934/) which reworked some of this logic. Unfortunately I'm not too familiar with this area of the codebase and our DevTools expert is out this week. It will need a little more investigation to figure out what's going wrong.

I only saw the issue happen when loading / refreshing the actual web page while Angular DevTools is already open. Once the extension encounters this broken state, it seems to be stuck there and refreshing the web page causes Angular DevTools to retry, but never succeeds.

**Workaround:** I found that right clicking on the Angular DevTools panel and selecting "Reload frame" fixes the problem, probably because the inspected web page is already loaded and avoids the race condition.
